### DISCUSSION TITLE
Translating to/from World coordinates?

### DISCUSSION THREAD

**Discussion by shanesveller** [StarArawn/bevy_ecs_tilemap#156] [Q&A] [answered]
Good morning, and thanks for the awesome crate! Does `bevy_ecs_tilemap` have any functionality to translate world coords to tilemap coords? At least for the trivial top-down squares style? I see some chatter in the issues that things like hexagons/isometric pose significant challenges, which I can easily believe.

**Comment by MrGVSV** [comment, marked as answer]
If your map's origin is (0.0, 0.0) and you're only concerned about basic top-down square tiles, then you should be able to just divide your world coordinates by your tile size (rounded down).

```rust
let tile_size = 32.0;
let tile_pos = (coords / tile_size).as_uvec();
```

For a tile size of `32.0`, this should map like:

| World        | Tile   |
| :----------: | :----: |
| (0.0, 0.0)   | (0, 0) |
| (32.0, 0.0)  | (1, 0) |
| (63.0, 0.0)  | (1, 0) |
| (65.0, 32.0) | (2, 1) |

---

Edit: Sorry math was completely wrong lol, should be fixed now.

**Reply by shanesveller** [reply]
Appreciate it, will mark as answered when I get a chance to test this. Thanks!

**Reply by StarArawn** [reply]
when you call `as_uvec` I believe it just drops the decimal numbers instead of rounding or flooring. In my opinion you should probably floor the coordinates as well, but you might find round or ceil to work better depending on your use case.

**Comment by bzm3r** [comment]
@shanesveller We have now provided a whole bunch of stuff to handle calculating tile position. See: https://github.com/StarArawn/bevy_ecs_tilemap/blob/eabd66eac4c5204ec54646141ffb26935cace88d/examples/mouse_to_tile.rs#L372

And: https://github.com/StarArawn/bevy_ecs_tilemap/blob/eabd66eac4c5204ec54646141ffb26935cace88d/examples/mouse_to_tile.rs#L257
